apache/bookkeeper
# Issue 339: Use ephemeral port for bookkeeper tests and variable unit test improvements

Descriptions of the changes in this PR:

- Enable ephemeral port by default for all BookKeeperClusterTestCase based test cases
- Removed Multi* based classes. A lot of unit tests unnecessarily extended from those base classes. Converted them to extend BookKeeperClusterTestCase directly. Confirmed this with @sijie 
- Reduce sleep period for some test cases

LedgerEntry fixes for #317  
- use checkState rather than checkNull to keep the behavior consistent as before
- fix BookKeeperTest#testReadEntryReleaseByteBufs 

TestBackwardCompat:
- removed testCompact400 (400 has been very long time ago, no one is running 400)
- BookKeeperNettyServer should shut down netty server when failed to construct a bookie

TestDiskChecker:
- increased the threshold to 0.99f. be consistent with TestBKConfiguration

BookieInitializationTest
- make each test method using different ledgers path, so registration won't be conflicting between tests

BookieJournalRollingTest
- make sure bookie port is used correctly on restarting bookies.
